src/kdf.rs

```diff
@@ -747,12 +747,13 @@ impl Mechanism for PBKDF2Mechanism {
                 if params.pPassword == std::ptr::null_mut()
                     || params.ulPasswordLen == 0
                 {
-                    return err_rv!(CKR_MECHANISM_PARAM_INVALID);
+                    self.mock_password_object(vec![0u8; 0])?
+                } else {
+                    self.mock_password_object(bytes_to_vec!(
+                        params.pPassword,
+                        params.ulPasswordLen
+                    ))?
                 }
```

Comment:
With change from https://github.com/latchset/kryoptic/pull/61/commits/0f6ec1842727585ef2248b24f973522da7e40ba1 you could squash this back to
```
                self.mock_password_object(bytes_to_vec!(
                    params.pPassword,
                    params.ulPasswordLen
                ))?
```